lib/pages/home_page.dart

```diff
@@ -610,10 +578,83 @@ class _HomePageState extends State<HomePage> with TickerProviderStateMixin {
     }
   }
 
+  void _onProjectSettingsChanged() {
+    _loadSettingsFromPrefs();
+    _saveProjectSettingsToFile(_projectDir!);
+  }
+
+  void _loadSettingsFromPrefs() {
+    setState(() {
+      _robotSize = Size(
+        widget.prefs.getDouble('robotWidth') ?? 0.75,
+        widget.prefs.getDouble('robotLength') ?? 1.0,
+      );
+      _holonomicMode = widget.prefs.getBool('holonomicMode') ?? false;
+      _generateJSON = widget.prefs.getBool('generateJSON') ?? false;
+      _generateCSV = widget.prefs.getBool('generateCSV') ?? false;
+      _pplibClient = widget.prefs.getBool('pplibClient') ?? false;
+
+      if (_pplibClient) {
+        PPLibClient.initialize(widget.prefs);
+      } else {
+        PPLibClient.stopServer();
+      }
+    });
+  }
+
+  void _loadProjectSettingsFromFile(Directory projectDir) {
+    File settingsFile = File(join(projectDir.path, _settingsDir));
+
+    if (settingsFile.existsSync()) {
+      try {
+        final fileContents = settingsFile.readAsStringSync();
```

Comment:
Since nothing from the settings file is actually needed immediately, you could make this method async, then use await to replace the sync methods with the async versions. This would stop this method from holding up initState and freezing the app while loading the file.

```suggestion
  void _loadProjectSettingsFromFile(Directory projectDir) async {
    File settingsFile = File(join(projectDir.path, _settingsDir));

    if (await settingsFile.exists()) {
      try {
        final fileContents = await settingsFile.readAsString();
```

Reply:
Gotcha! I was in between, but I agree they shouldn't be needed right away, so async should be fine here